core/src/main/java/org/elasticsearch/common/network/NetworkModule.java

```diff
@@ -141,22 +170,36 @@ public AllocationCommandRegistry getAllocationCommandRegistry() {
         return namedWriteables;
     }
 
-    @Override
-    protected void configure() {
-        bind(NetworkService.class).toInstance(networkService);
-        bindTransportService();
-        transportTypes.bindType(binder(), settings, TRANSPORT_TYPE_KEY, TRANSPORT_DEFAULT_TYPE_SETTING.get(settings));
-        bind(TransportInterceptor.class).toInstance(new CompositeTransportInterceptor(this.transportIntercetors));
-        if (transportClient == false) {
-            if (HTTP_ENABLED.get(settings)) {
-                bind(HttpServer.class).asEagerSingleton();
-                httpTransportTypes.bindType(binder(), settings, HTTP_TYPE_SETTING.getKey(), HTTP_DEFAULT_TYPE_SETTING.get(settings));
-            } else {
-                bind(HttpServer.class).toProvider(Providers.of(null));
-            }
-            // Bind the AllocationCommandRegistry so RestClusterRerouteAction can get it.
-            bind(AllocationCommandRegistry.class).toInstance(allocationCommandRegistry);
+    public NetworkPlugin.TransportFactory<HttpServerTransport> getHttpServerTransportFactory() {
+        final String name;
+        if (HTTP_TYPE_SETTING.exists(settings)) {
+            name = HTTP_TYPE_SETTING.get(settings);
+        } else {
+            name = HTTP_DEFAULT_TYPE_SETTING.get(settings);
+        }
+        final NetworkPlugin.TransportFactory<HttpServerTransport> factory = transportHttpFactories.get(name);
+        if (factory == null) {
+            throw new IllegalStateException("No http.type: " + HTTP_TYPE_SETTING.get(settings) + " configured");
```

Comment:
This comes up if you configure the transport to use some key that we don't have an implementation for, right? Maybe say something like `"Unsupported http.type [" +name + "]"`.?